lib/ocpp/v16/charge_point_impl.cpp

```diff
@@ -2372,8 +2370,8 @@ void ChargePointImpl::handleClearChargingProfileRequest(ocpp::Call<ClearCharging
                this->smart_charging_handler->clear_all_profiles_with_filter(
                    call.msg.id, call.msg.connectorId, call.msg.stackLevel, call.msg.chargingProfilePurpose, true)) {
         response.status = ClearChargingProfileStatus::Accepted;
-
-    } else if (this->smart_charging_handler->clear_all_profiles_with_filter(
+    } else if (!call.msg.id and
+               this->smart_charging_handler->clear_all_profiles_with_filter(
                    call.msg.id, call.msg.connectorId, call.msg.stackLevel, call.msg.chargingProfilePurpose, false)) {
```

Comment:
since `call.msg.id` must be std::nullopt (because of the`!call.msg.id` in the if) at this point maybe you can directly pass `std::nullopt for `call.msg.if` here which would make this a bit more readable